runtime/precompiles/src/staking.rs

```diff
@@ -0,0 +1,348 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Precompile to call parachain-staking runtime methods via the EVM
+use evm::{Context, ExitError, ExitSucceed};
+use frame_support::dispatch::{Dispatchable, GetDispatchInfo, PostDispatchInfo};
+use frame_support::traits::Currency;
+use pallet_evm::AddressMapping;
+use pallet_evm::GasWeightMapping;
+use pallet_evm::Precompile;
+use sp_core::H160;
+use sp_core::U256;
+use sp_std::convert::TryFrom;
+use sp_std::convert::TryInto;
+use sp_std::fmt::Debug;
+use sp_std::{marker::PhantomData, vec::Vec};
+
+type BalanceOf<Runtime> = <<Runtime as parachain_staking::Config>::Currency as Currency<
+	<Runtime as frame_system::Config>::AccountId,
+>>::Balance;
+
+/// A precompile to wrap the functionality from parachain_staking.
+///
+/// EXAMPLE USECASE:
+/// A simple example usecase is a contract that allows donors to donate, and stakes all the funds
+/// toward one fixed address chosen by the deployer.
+/// Such a contract could be deployed by a collator candidate, and the deploy address distributed to
+/// supporters who want to donate toward a perpetual nomination fund.
+pub struct ParachainStakingWrapper<Runtime>(PhantomData<Runtime>);
+
+impl<Runtime> Precompile for ParachainStakingWrapper<Runtime>
+where
+	Runtime: parachain_staking::Config + pallet_evm::Config,
+	Runtime::AccountId: From<H160>,
+	BalanceOf<Runtime>: TryFrom<U256> + Debug,
+	Runtime::Call: Dispatchable<PostInfo = PostDispatchInfo> + GetDispatchInfo,
+	<Runtime::Call as Dispatchable>::Origin: From<Option<Runtime::AccountId>>,
+	Runtime::Call: From<parachain_staking::Call<Runtime>>,
+{
+	fn execute(
+		input: &[u8], //Reminder this is big-endian
+		target_gas: Option<u64>,
+		context: &Context,
+	) -> Result<(ExitSucceed, Vec<u8>, u64), ExitError> {
+		log::info!("In parachain staking wrapper");
```

Comment:
```suggestion
		log::trace!(target: "staking-precompile", "In parachain staking wrapper");
```

And the same thing in all the other places. Otherwise the node log will be super noisy.

Reply:
Do we need this? These log statements inside the runtime/EVM have been known to have enormous overhead.

Reply:
...or does this effectively disable them?

```toml
on-chain-release-build = ["sp-api/disable-logging"]
```

Reply:
The performance issue is that the logs are generated whether they are printed or not. But yes @notlesh that `on-chain-release-build` feature is for exactly this.